clvm_tools/cmds.py

```diff
@@ -192,11 +198,20 @@ def launch_tool(args, tool_name, default_stage=0):
             time_assemble = time.perf_counter()
 
         input_sexp = to_sexp_f((assembled_sexp, env))
-        time_parse_input = time.perf_counter()
-        cost, result = run_program(
-            run_script, input_sexp, max_cost=args.max_cost, pre_eval_f=pre_eval_f)
-        time_done = time.perf_counter()
 
+        if args.backend == "rust" or (serialize_and_run_program and args.backend != "python"):
+            run_script = run_script.as_bin()
+            input_sexp = input_sexp.as_bin()
+            time_parse_input = time.perf_counter()
+            cost, result = serialize_and_run_program(
+                run_script, input_sexp, 1, 3, args.max_cost)
+            time_done = time.perf_counter()
```

Comment:
Isn't it kind of cheating not including the deserialization in the time?

Reply:
well, I don't think so. I sort of consider it the responsibility of the caller to interpret the result. If juggling a tree of small heap objects is too slow, the caller can do it more efficiently